backend/graph/messages.go

```diff
@@ -0,0 +1,62 @@
+package graph
+
+import (
+	"context"
+	cache "github.com/Code-Hex/go-generics-cache"
+	"github.com/ansel1/merry/v2"
+	"github.com/bcc-code/brunstadtv/backend/common"
+	"sync"
+	"time"
+)
+
+type timedCacheEntry struct {
+	Cached time.Time
+	Entry  any
+}
+
+var messageCache = cache.New[string, timedCacheEntry]()
+
+var messageLock = sync.Mutex{}
+
+func getMessagesFromCache(timestamp *time.Time) ([]common.MaintenanceMessage, bool) {
+	if messages, ok := messageCache.Get("maintenance"); ok && (timestamp == nil || messages.Cached.Equal(*timestamp) || messages.Cached.After(*timestamp)) {
+		return messages.Entry.([]common.MaintenanceMessage), true
+	}
+	return nil, false
+}
+
+var truncateTimeInterval time.Duration = 1
+
+func (r *Resolver) getMaintenanceMessages(ctx context.Context, timestamp *time.Time) ([]common.MaintenanceMessage, error) {
+	// Truncate time to avoid updating again within seconds
+	if timestamp != nil {
+		if timestamp.After(time.Now()) {
+			return nil, merry.New("invalid time (future)")
```

Comment:
```suggestion
			timestamp = time.now()
```
We don't really "care" if people attempt to send in junk, so let's still return something reasonably sensible.